libcaf_net/test/detail/rfc6455.cpp

```diff
@@ -96,6 +130,44 @@ CAF_TEST(valid mask key plus small data) {
   CAF_CHECK_EQUAL(hdr.payload_len, data.size());
 }
 
+CAF_TEST(no mask key plus upper bound on small data) {
+  std::vector<uint8_t> data;
+  data.resize(125, 0xFF);
+  byte_buffer out;
+  impl::assemble_frame(impl::binary_frame, 0, as_bytes(make_span(data)), out);
+  CAF_CHECK_EQUAL(take(out, 6), bytes({
+                                  0x82, // FIN + binary frame opcode
+                                  0x7D, // data size = 125
+                                  0xFF, 0xFF, 0xFF, 0xFF, // masked data
+                                }));
+  impl::header hdr;
+  CAF_CHECK_EQUAL(impl::decode_header(out, hdr), 2);
+  CAF_CHECK_EQUAL(hdr.fin, true);
+  CAF_CHECK_EQUAL(hdr.mask_key, 0u);
+  CAF_CHECK_EQUAL(hdr.opcode, impl::binary_frame);
+  CAF_CHECK_EQUAL(hdr.payload_len, data.size());
+}
+
+CAF_TEST(valid mask key plus upper bound on small data) {
+  std::vector<uint8_t> data;
+  data.resize(125, 0xFF);
+  byte_buffer out;
+  impl::assemble_frame(impl::binary_frame, 0xDEADC0DE,
+                       as_bytes(make_span(data)), out);
+  CAF_CHECK_EQUAL(take(out, 10), bytes({
+                                   0x82, // FIN + binary frame opcode
+                                   0xFD, // MASKED + data size = 125
+                                   0xDE, 0xAD, 0xC0, 0xDE, // mask key,
+                                   0xFF, 0xFF, 0xFF, 0xFF, // masked data
+                                 }));
+  impl::header hdr;
+  CAF_CHECK_EQUAL(impl::decode_header(out, hdr), 6);
+  CAF_CHECK_EQUAL(hdr.fin, true);
+  CAF_CHECK_EQUAL(hdr.mask_key, 0xDEADC0DEu);
```

Comment:
```suggestion
  CAF_CHECK_EQUAL(hdr.mask_key, 0xDEADC0DE);
```

Integers with hexcode notation are always unsigned:

```
    using type_t = decltype(0xDEADC0DE);
    static_assert(std::is_same_v<type_t, unsigned>);
```